### PR TITLE
Add vp9 svc support by Dependency Descriptor

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: pkg/sfu/forwarder.go [hunk 1]
```diff
@@ -570,14 +570,14 @@ func (f *Forwarder) AllocateOptimal(availableLayers []int32, brs Bitrates, allow
 					alloc.TargetLayers.Spatial = l
 				}
 			}
-			alloc.TargetLayers.Temporal = buffer.DefaultMaxLayerTemporal
+			alloc.TargetLayers.Temporal = f.maxLayers.Temporal
```
Comment (cnderrauber) [inline pkg/sfu/forwarder.go:573]: This change to fix client preferred fps (temporary layer), can you verify it is correct? @boks1971

Comment (boks1971) [reply pkg/sfu/forwarder.go:573]: It should be okay @cnderrauber , but we probably need to allow that opportunistic higher layer. I will looking at adding it back for simulcast tracks and use proper layer for DD tracks.

File: pkg/sfu/buffer/helpers.go [hunk 2]
```diff
@@ -351,4 +353,28 @@ func IsAV1Keyframe(payload []byte) bool {
...
+
+	if (vp9.Payload[0] & 0xc0) != 0x80 {
+		return false
+	}
+
+	profile := (vp9.Payload[0] >> 4) & 0x3
+	if profile != 3 {
+		return (vp9.Payload[0] & 0xC) == 0
+	}
+	return (vp9.Payload[0] & 0x6) == 0
```
Comment (boks1971) [inline pkg/sfu/buffer/helpers.go:377]: Don't fully understand this. I cannot read the bit stream format document :-). But, this RFC seems to say that we do not need to dig beyond the VP9 Payload Descriptor to determine key frame. I may change this function based on that when I do my refactor. Guess, we really do not need to know if key frame or not with dependency descriptor. We will need that only when using simulcast.

https://datatracker.ietf.org/doc/html/draft-ietf-payload-vp9-16. The actual text
```
      A key picture is a picture whose base
      spatial layer frame is a key frame, and which thus completely
      resets the encoder state.  This packet will have its P bit equal
      to zero, SID or L bit (described below) equal to zero, and B bit
      (described below) equal to 1.
```
This is what I used in the refactor branch I am working on. I will compare with the above and check if they match.

File: pkg/sfu/forwarder.go [hunk 3]
```diff
@@ -570,14 +570,14 @@ func (f *Forwarder) AllocateOptimal(availableLayers []int32, brs Bitrates, allow
...
 
 			alloc.RequestLayerSpatial = alloc.TargetLayers.Spatial
 		} else {
 			requestLayerSpatial := int32(math.Min(float64(f.maxLayers.Spatial), float64(f.maxPublishedLayer)))
 			if f.currentLayers.IsValid() && requestLayerSpatial == f.requestLayerSpatial && f.currentLayers.Spatial == f.requestLayerSpatial {
 				// current is locked to desired, stay there
-				alloc.TargetLayers = f.currentLayers
+				alloc.TargetLayers = buffer.VideoLayer{Spatial: f.requestLayerSpatial, Temporal: f.maxLayers.Temporal}
 				alloc.RequestLayerSpatial = f.requestLayerSpatial
```
Comment (boks1971) [inline pkg/sfu/forwarder.go:581]: Might have to revert these changes back to how they were and deal with layer limiting based on DD or not. But, I will take care of it in the refactor branch.

**Review by boks1971 — APPROVED** [review]

**Pull request merged**